doc: fix "documented empty return type of..." warnings

dcpleung opened the pull request:
There are bunch of "documented empty return type of ..." warnings coming from doxygen when parsing function documentation. This is due to `@return` being specified for void functions. So remove these to fix the warnings.

Example warning:
```
WARNING: zephyr/rtos/zephyr/include/drivers/modem/gsm_ppp.h:40: documented empty return type of gsm_ppp_register_modem_power_callback
WARNING: zephyr/rtos/zephyr/include/sys/sys_heap.h:204: documented empty return type of sys_heap_stress
```

dcpleung committed:
kernel: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
sys: include: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
drivers: interrupt_controller: remove @return doc for void funcs

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
drivers: timer: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
drivers: console: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
drivers: serial: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
drivers: cache: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
drivers: i2s/i2s_litex: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
usb: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
lib: os: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
bluetooth: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
cpp: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
random: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
net: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
tfm: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
drivers: pcie: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
drivers: ec_host_cmd_periph: remove @return doc for void funcs

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
riscv: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
nios2: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
posix: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
sparc: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
xtensa: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
doc: remove @return doc for void functions in sample code

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
x86: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
arc: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
arm: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
tests: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
audio: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
shell: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
modbus: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
mcumgr: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
modem: remove @return doc for void functions

For functions returning nothing, there is no need to document
with @return, as Doxgen complains about "documented empty
return type of ...".

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
tests: mslab_api: remove double asterisk inside func comments

Doxygen treats "/**" as documentation even if the comment block
is inside the function. So remove the extra asterisk for those
"TESTPOINT" comments.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

gmarull reviewed (approved):
Thanks for these fixes. I don't remember Doxygen failing when run in warnings as error mode, is this warning type disabled by default?

dcpleung commented:
> Thanks for these fixes. I don't remember Doxygen failing when run in warnings as error mode, is this warning type disabled by default?

There doesn't seem to be a config parameter for this particular warning. Maybe `WARN_IF_DOC_ERROR`? But it is already set to `YES` in our config file.

stephanosio commented on arch/arm/core/aarch32/cortex_m/fault.c:
```diff
@@ -353,7 +353,6 @@ static uint32_t mem_manage_fault(z_arch_esf_t *esf, int from_hard_fault,
  * See z_arm_fault_dump() for example.
  *
- * @return N/A
  */
 static int bus_fault(z_arch_esf_t *esf, int from_hard_fault, bool *recoverable)
```
This was incorrectly documented as `@return N/A`. It needs to be:
```
@return error code to identify the fatal error reason
```

stephanosio commented on drivers/interrupt_controller/intc_ioapic.c:
```diff
@@ -153,12 +153,9 @@ static bool get_vtd(void)
- * @return N/A
  */
 __boot_func
 int ioapic_init(const struct device *unused)
```
another one here

stephanosio commented on drivers/interrupt_controller/intc_plic.c:
```diff
@@ -120,14 +113,12 @@ void riscv_plic_set_priority(uint32_t irq, uint32_t priority)
- * @return N/A
+ * @param irq IRQ number for which to set priority
  */
 int riscv_plic_get_irq(void)
```
ditto

stephanosio commented on drivers/interrupt_controller/intc_plic.c:
```diff
@@ -174,9 +165,7 @@ static void plic_irq_handler(const void *arg)
  * @brief Initialize the Platform Level Interrupt Controller
- * @return N/A
  */
 static int plic_init(const struct device *dev)
```
ditto

stephanosio commented on drivers/timer/arcv2_timer0.c:
```diff
@@ -108,54 +107,39 @@ static ALWAYS_INLINE uint32_t timer0_count_register_get(void)
- *
- * @return N/A
  */
 static ALWAYS_INLINE uint32_t timer0_control_register_get(void)
```
ditto

stephanosio commented on drivers/timer/arcv2_timer0.c:
```diff
@@ -108,54 +107,39 @@ static ALWAYS_INLINE uint32_t timer0_count_register_get(void)
- *
- * @return N/A
  */
 static ALWAYS_INLINE uint32_t timer0_limit_register_get(void)
```
ditto

stephanosio commented on kernel/mem_slab.c:
```diff
@@ -20,8 +20,6 @@
- *
- * @return N/A
  */
 static int create_free_list(struct k_mem_slab *slab)
```
ditto

stephanosio commented on kernel/mem_slab.c:
```diff
@@ -49,8 +47,6 @@ static int create_free_list(struct k_mem_slab *slab)
- *
- * @return N/A
  */
 static int init_mem_slab_module(const struct device *dev)
```
ditto

stephanosio commented on soc/arm/nxp_lpc/lpc54xxx/soc.c:
```diff
@@ -144,7 +142,7 @@ static const char core_m0[] = {
- * @return N/A
+ *
  */
 /* This function is also called at deep sleep resume. */
 int _slave_init(const struct device *arg)
```
ditto

stephanosio commented on soc/arm/nxp_lpc/lpc55xxx/soc.c:
```diff
@@ -228,7 +226,7 @@ SYS_INIT(nxp_lpc55xxx_init, PRE_KERNEL_1, 0);
- * @return N/A
+ *
  */
 /* This function is also called at deep sleep resume. */
 int _second_core_init(const struct device *arg)
```
ditto

stephanosio commented on subsys/cpp/cpp_dtors.c:
```diff
@@ -23,8 +23,6 @@ __weak void *__dso_handle;
- *
- * @return N/A
  */
 int __cxa_atexit(void (*destructor)(void *), void *objptr, void *dso)
```
ditto

`@return 0 on success`

stephanosio commented on subsys/usb/class/cdc_acm.c:
```diff
@@ -740,8 +724,6 @@ static void cdc_acm_baudrate_set(const struct device *dev, uint32_t baudrate)
- *
- * @return  N/A.
  */
 static int cdc_acm_send_notification(const struct device *dev,
```
ditto

bbolen commented on drivers/interrupt_controller/intc_plic.c:
```diff
@@ -98,15 +93,13 @@ int riscv_plic_irq_is_enabled(uint32_t irq)
- * @param irq IRQ number for which to set priority
  *
- * @return N/A
+ * @param irq IRQ number for which to set priority
```
Since you added a line for irq, can you add one for priority as well?

dcpleung replied:
Done.

bbolen commented on drivers/serial/uart_nrfx_uart.c:
```diff
@@ -191,8 +191,6 @@ static void event_txdrdy_clear(void)
  * @param dev UART device struct
  * @param baudrate Baud rate
- *
- * @return N/A
```
you might want to note that this function returns -EINVAL for a bad baudrate, 0 on success.

bbolen replied:
I got a request for a re-review, but I don't see where my comments were addressed for this or intc_plic.c

dcpleung replied:
Sorry. Missed this one. Fixed now.

de-nordic reviewed (approved):
mgmt/mcumgr ok.

carlocaione reviewed (approved):
For ARM64 / cache

chrta commented on drivers/interrupt_controller/intc_gic.c:
```diff
@@ -220,11 +220,7 @@ static void gic_cpu_init(void)
- *
  * @brief Initialize the GIC device driver
- *
- *
- * @return N/A
  */
```
This block should probably be placed right before the function `arm_gic_init()` (new line 228) and not before the defines.

dcpleung committed:
i2s: i2s_litex: fix doc styling on two copy functions

Just minor styling changes to capitalize the first character
and removing the trailing comma on the brief doc description
on two copy functions.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung commented:
CI failure is due to ztress, and is not related to doc changes.
